src/wallet/test/wallet_tests.cpp

```diff
@@ -329,6 +329,39 @@ BOOST_FIXTURE_TEST_CASE(importwallet_rescan, TestChain100Setup)
     }
 }
 
+// This test verifies that wallet settings can be added and removed
+// Concurrently, ensuring no race conditions occur during either process.
+BOOST_FIXTURE_TEST_CASE(write_wallet_settings_concurrently, TestingSetup)
+{
+    WalletContext context;
+    context.chain = m_node.chain.get();
+    const int NUM_WALLETS = 5;
+    std::vector<std::thread> threads;
+    {
+        for (int i{0}; i < NUM_WALLETS; i++) {
+            threads.emplace_back([i, &context] {
+                BOOST_CHECK_MESSAGE(AddWalletSetting(context, strprintf("wallet_%d", i)), strprintf("write wallet_%d failed", i));
```

Comment:
I suppose I should update to check for the boolean returned value of true not a string message.

 I will update to just `Assert` the returned value is true.

 I will test your suggestion, thanks

Reply:
As per https://github.com/bitcoin/bitcoin/pull/30697#discussion_r1727410617, I'm not sure capturing the offending thread adds value, since it won't be deterministic anyway? And we're already testing that all wallets were added/removed successfully? Also no objection to adding an `Assert`, though.

Reply:
Done thanks.